pyhms/cluster/cluster.py

```diff
@@ -0,0 +1,62 @@
+import numpy as np
+
+from ..core.population import Population
+from ..demes.cma_deme import CMADeme
+
+
+def mahalanobis_distance(x: np.ndarray, y: np.ndarray, covariance_matrix_inverse: np.ndarray) -> float:
+    diff = x - y
+    return np.sqrt(diff @ covariance_matrix_inverse @ diff)
+
+
+DEFAULT_CLUSTER_SIZE = 30
+
+
+class Cluster:
+    def __init__(
+        self,
+        population: Population,
+        mean: np.ndarray,
+        covariance_matrix: np.ndarray,
+    ):
+        self.population = population
+        self.mean = mean
+        self.covariance_matrix = covariance_matrix
+        self.covariance_matrix_inverse = np.linalg.inv(covariance_matrix)
+
+    @classmethod
+    def from_deme(cls, deme: CMADeme) -> "Cluster":
+        return cls(
+            Population.from_individuals(deme.all_individuals[-DEFAULT_CLUSTER_SIZE:]),
+            deme.mean,
+            deme.covariance_matrix,
+        )
```

Comment:
I like this alternative utility constructor. One think, that I'm not sure of, is that it currently works only with CMADeme and the name does not suggest it. If used with something different, we would get an exception. I see, that it's guarded from this in LandscapeApproximator. See my comment there.

Reply:
ok, I will improve the name